metaflow/plugins/kfp/s3_sensor_decorator.py

```diff
@@ -0,0 +1,85 @@
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+
+from types import FunctionType
+from typing import Tuple
+from urllib.parse import urlparse
+
+"""
+Within identity_formatter, which is passed in as the path_formatter parameter,
+customers have access to all variables in flow_parameters_json (which 
+include all parameters passed in through the flow at compile and run time by
+KFP) as well as a number of environment variables.
+
+Please see metaflow/plugin/environment_decorator.py for details on how
+to add environment variables to be accessible within steps.
+"""
+
+
+def identity_formatter(path: str, flow_parameters: dict) -> str:
+    return path
+
+
+"""
+@s3_sensor is implemented as a flow decorator that's used by customers to ensure 
+a workflow only begins after a key in S3 has been written to. 
+
+Example usage:
+
+    If FLOW_ID is a parameter you've passed to your flow, the substitution
+    of FLOW_ID in the `path` variable is automatically done for you if you
+    specify the variable in braces ({}) (e.g. {FLOW_ID}).
+
+    @s3_sensor(
+        path="s3://aip-example-sandbox/metaflow/S3SensorFlow/data/61/{FLOW_ID}",
+        timeout_seconds=3600, # 1 hour
+        polling_interval_seconds=90,
+    )
+    class S3SensorFlow(FlowSpec):    
+        ...
+
+    If you want to format FLOW_ID (or any other variable in flow_parameters), you
+    can do so with a separate `path_formatter` function:
+
+    def formatter(path: str, flow_parameters: dict) -> str:
+        return path.format(FLOW_ID, my_parse_func(flow_parameters["FLOW_ID"]))
+
+    @s3_sensor(
+        path="s3://aip-example-sandbox/metaflow/S3SensorFlow/data/61/{FLOW_ID}",
```

Comment:
Can you have an example of another format variable, rather than FLOW_ID which is already a parameter passed into the Flow? Else this becomes confusing.

Also, pythonic style would have variables lower cased.